smithy-model/src/test/java/software/amazon/smithy/model/transform/DowngradeToV1Test.java

```diff
@@ -145,10 +152,38 @@ public void removesUnnecessaryDefaults() {
         assertThat(downgraded.expectShape(integerShape).hasTrait(DefaultTrait.class), Matchers.is(false));
         assertThat(downgraded.expectShape(integerShape).hasTrait(BoxTrait.class), Matchers.is(true));
 
+        // ZeroInteger keeps the default trait since it matches the zero value of a 1.0 model
+        ShapeId zeroIntegerShape = ShapeId.from("smithy.example#ZeroInteger");
+        assertThat(downgraded.expectShape(zeroIntegerShape).hasTrait(DefaultTrait.class), Matchers.is(true));
+        assertThat(downgraded.expectShape(zeroIntegerShape).hasTrait(BoxTrait.class), Matchers.is(false));
+
+        ShapeId boxedIntegerShape = ShapeId.from("smithy.example#BoxedInteger");
+        assertThat(downgraded.expectShape(boxedIntegerShape).hasTrait(DefaultTrait.class), Matchers.is(false));
+        assertThat(downgraded.expectShape(boxedIntegerShape).hasTrait(BoxTrait.class), Matchers.is(true));
+
         StructureShape dStruct = downgraded.expectShape(ShapeId.from("smithy.example#Struct"), StructureShape.class);
         assertThat(dStruct.getAllMembers().get("foo").hasTrait(DefaultTrait.class), Matchers.is(false));
+        assertThat(dStruct.getAllMembers().get("foo").hasTrait(BoxTrait.class), Matchers.is(false));
+
+        assertThat(dStruct.getAllMembers().get("defaultZero").hasTrait(DefaultTrait.class), Matchers.is(false));
```

Comment:
`// ZeroInteger keeps the default trait since it matches the zero value of a 1.0 model` above and this seems inconsistent. It is implying we retain default zero value for root shape but not member - is this expected behavior?

I see `// Members with a null default are considered boxed. Keep the trait to retain consistency with other indexes and checks.` in the source code for `null` default to be retained. But doesn't say about retain default zero on root/member. _If_ the current behavior is expected, we should at least explain that in the code comment as it is not intuitive.

Reply:
This will be fixed in the next revision to keep the default zero value for members that have a default zero value and a target shape that has a default zero value.